Std/Data/HashMap/Basic.lean

```diff
@@ -344,13 +448,35 @@ instance : GetElem (HashMap α β) α (Option β) fun _ _ => True where
     | none => map.insert k v₂
     | some v₁ => map.insert k $ f k v₁ v₂
 
-/-- Runs a monadic function over the elements in the map (in arbitrary order). -/
+/--
+Runs a monadic function over the elements in the map (in arbitrary order).
+```
+def checkEven (k : String) (v : Nat) : Except String Unit :=
+  if v % 2 == 0 then pure () else throw s!"value {v} at key {k} is not even"
+
+forM checkEven (ofList [("one", 1), ("three", 3)]) = Except.error "value 3 at key three is not even"
+forM checkEven (ofList [("two", 2), ("four", 4)]) = Except.ok ()
+```
+-/
 @[inline] def forM [Monad m] (f : α → β → m PUnit) (self : HashMap α β) : m PUnit := self.1.forM f
 
-/-- Converts the map into a list of key-value pairs. -/
+/--
+Converts the map into a list of key-value pairs.
+```
+TODO: confirm this acceptable
+def hashMap := ofList [("one", 1), ("two", 2)]
+hashMap.toList = [("one", 1), ("two", 2)] ∨ hashMap.toList = [("two", 2), ("one", 1)]
```

Comment:
Is it acceptable to use `∨` to demonstrate the order of the list isn't deterministic?

Reply:
How about just asserting that it's a permutation of `[("one", 1), ("two", 2)]`?

Reply:
That is much more clear. I have updated this example (and similarly the `toArray` example) to `(ofList [("one", 1), ("two", 2)]).toList ∈ List.permutations [("one", 1), ("two", 2)]`

Reply:
List.Perm has a syntax `~` which is available if you `open List`, so you could also write `(ofList [("one", 1), ("two", 2)]).toList ~ [("one", 1), ("two", 2)]`

Reply:
Ah, nice. Updated `toList` and `toArray` examples with the `~` syntax